### PR TITLE
SAPIG-237: Add Affinity codeblock to SAPIG components

### PR DESCRIPTION
**Pull request by shaunharrisonFR**
Add in affinity codeblock but have way of enabling and disabling, will be disabled by default as not required in dev

Issue: https://bugster.forgerock.org/jira/browse/SAPIG-237

### CODE REVIEW

File: _infra/helm/securebanking-openbanking-uk-rcs/templates/deployment.yaml [hunk 1]
```diff
@@ -14,6 +14,18 @@ spec:
       maxSurge: {{ .Values.deployment.rollingUpdate.maxSurge }}
       maxUnavailable: {{ .Values.deployment.rollingUpdate.maxUnavailable }}
     {{ end }}
+  {{ if eq .Values.deployment.useAffinity "enabled" }}
+  affinity:
+    podAntiAffinity:
```
Comment (dbadham-fr) [inline _infra/helm/securebanking-openbanking-uk-rcs/templates/deployment.yaml:19]: Why antiAffinity?

Seems quite a specify set of rules to opt into, would it be better to allow a whole block of yaml to configure affinity to be dropped in?

**Review by dbadham-fr — CHANGES_REQUESTED** [review]

**Comment by shaunharrisonFR** [comment]
From my understanding of how this works, we don't want another RCS pod being deployed on a node that already has RCS running. 

Here is an example of anti affinity
https://docs.openshift.com/container-platform/3.11/admin_guide/scheduling/pod_affinity.html#admin-guide-sched-affinity-pod-config


But I do agree that it is specific so yeah will rework it

**Comment by shaunharrisonFR** [comment]
Added the values in for now to make sure it's applied correctly in the PR build (looking at the indentation) 

If it's okay i'll change it around to be empty

**Review by BohoCode — APPROVED** [review]

**Review by BohoCode — COMMENTED** [review]
Will also need documenting in the deployment docs

**Comment by shaunharrisonFR** [comment]
image
Added to the documentation for customer specification

**Review by dbadham-fr — APPROVED** [review]

File: _infra/helm/securebanking-openbanking-uk-rcs/values.yaml [hunk 2]
```diff
@@ -1,4 +1,7 @@
 deployment:
+  # Affinity Configuration
+  # Uncomment and add in required configuration 
+  # affinity:
```
Comment (dbadham-fr) [inline _infra/helm/securebanking-openbanking-uk-rcs/values.yaml:4]: Have you tried:
```
affinity: {}
```

Example in nginx chart:
https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/values.yaml#L218C3-L218C15
https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/templates/default-backend-deployment.yaml#L116-L118

I'd rather not have commented out stuff where possible

**Review by dbadham-fr — CHANGES_REQUESTED** [review]

**Review by dbadham-fr — APPROVED** [review]

**Pull request merged**